firebase_admin/functions.py

```diff
@@ -0,0 +1,431 @@
+            return False
+
+
+@dataclass
+class TaskOptions:
+    """Task Options that can be applied to a Task.
+    Args:
+        schedule_delay_seconds: The duration of delay of the time when the task is scheduled to be
+            attemptedor retried. This delay is added to the current time. Should only been set if
+            ``schedeule_time``is not set.
+
+        schedule_time: The time when the task is scheduled to be attempted or retried. Should only
+            been set if ``schedule_delay_seconds`` is not set.
+
+        dispatch_deadline_seconds: The deadline for requests sent to the worker. If the worker does
+            not respond by this deadline then the request is cancelled and the attempt is marked as
+            a ``DEADLINE_EXCEEDED`` failure. Cloud Tasks will retry the task according to the
+            ``RetryConfig``. The default is 10 minutes. The deadline must be in the range of 15
+            seconds and 30 minutes (1800 seconds).
+
+        task_id: The ID to use for the enqueued event. If not provided, one will be automatically
+            generated.
+
+            If provided, an explicitly specified task ID enables task de-duplication.
+            Task IDs should be string that contain only letters ([A-Za-z]), numbers ([0-9]),
+            hyphens (-), or underscores (_) with a maximum length of 500 characters. If a task's
+            ID is identical to that of an existing task or a task that was deleted or executed
+            recently then the call will throw an error with code "functions/task-already-exists".
+            Another task with the same ID can't be created for ~1hour after the original task was
+            deleted or executed.
+
+            Because there is an extra lookup cost to identify duplicate task IDs, setting ID
+            significantly increases latency. Using hashed strings for the task ID or for the prefix
+            of the task ID is recommended. Choosing task IDs that are sequential or have sequential
+            prefixes, for example using a timestamp, causes an increase in latency and error rates
+            in all task commands. The infrastructure relies on an approximately uniform distribution
+            of task IDs to store and serve tasks efficiently.
+
+            "Push IDs" from the Firebase Realtime Database make poor IDs because they are based on
+            timestamps and will cause contention (slowdowns) in your task queue. Reversed push IDs
+            however form a perfect distribution and are an ideal key. To reverse a string in python
+            use ``reversedString = someString[::-1]``
+
+        headers: HTTP request headers to include in the request to the task queue function. These
+            headers represent a subset of the headers that will accompany the task's HTTP request.
+            Some HTTP request headers will be ignored or replaced, e.g. Authorization, Host,
+            Content-Length, User-Agent etc. cannot be overridden.
```

Comment:
```suggestion
            Some HTTP request headers will be ignored or replaced: `Authorization`, `Host`,
            `Content-Length`, `User-Agent` and others cannot be overridden.
```

Reply:
Also, a complete list would be better

Reply:
These are specifically referencing the the headers that Cloud Tasks ignores and replaces. Maybe adding a note to referencing it would be enough?

> A complete list of these ignored or replaced headers can be found in the following definition of the HttpRequest.headers property: https://cloud.google.com/tasks/docs/reference/rest/v2/projects.locations.queues.tasks#httprequest